### ISSUE TITLE
Playback not resuming from last position after restart

### ISSUE THREAD

**Issue by ZeiP** [gpodder/gpodder-sailfish#87]
When I partly listen to an episode and then restart the application, it correctly shows in the playback bar the last position. However, when I hit play, it starts from the beginning of the episode.

This seems similar to the issue in #17. The Jolla store version of the software is version 4.6.0 from mid-2015; should it already be fixed there?

**Comment by carmenfdezb** [comment]
I have the same issue but it doesn't always happens: sometimes works fine to me.

**Comment by Keeper-of-the-Keys** [comment]
I suspect that what happens is that upon loss of communication gpodder receives an EOF or null byte causing it to treat the stream as ended, it's something I have wanted to look at for a while.

The solution I currently have in mind is that podcasts even when the user does not choose to download them should be downloaded to a temporary file when they start being played instead of being streamed only. However since this may impact data usage maybe this should be a setting, either way since I haven't yet looked at the code that handles this this is all just speculation.

**Comment by Keeper-of-the-Keys** [comment]
Sorry I seem to be mixing two different issues, yes the resume for me also misbehaves, I need to look at the code the problem with it is that it doesn't seem to misbehave in a predictable manner which may make trouble shooting that much more difficult.

**Comment by ZeiP** [comment]
Yeah, I'm experiencing the issue always with downloaded episodes (since I don't bother streaming them directly ever because it too had some issues).

At least right now it's very predictable on my phone with the issue happening every time as described. I tried to download the source code and deploy the most recent version in SailfishOS IDE, but in that version I'm unable to even get anything to play, so I can't test if the most recent VCS code works better. It has some UI changes that make it better, so apparently the release version in both Harbour and Openrepos is quite old.

**Comment by Keeper-of-the-Keys** [comment]
Yeah we are still waiting for Jolla to respond, openrepos I haven't contacted yet.

**Comment by sfbg** [comment]
I am subscribed to about 20 episodes and 95% of the time it works as expected, apart from the Nothiing But the Blues Podcast, where it is almost certain, that after restart it will reset to the beginning:
http://cliff.libsyn.com/rss

Do you encounter the issue across the board?

**Comment by Keeper-of-the-Keys** [comment]
I did more testing on this issue, and I suspect this comes from report_playback_event() not being sent often enough/at the right times, what I have found is that it gets fired when the position in the episode is changed with the **previous** position (so if say I jumped from 2:00 to 5:15 then the position being stored will be 2:00 and only when I jump again or play/pause will another change be stored) it also does not seem to be fired when gpodder is closed to store the current playback position.

I think that at the very least on close the position should be updated and possible also at some regular time interval.
